### ISSUE TITLE
Segmentation fault caused on reading files [negative scenario]

### ISSUE THREAD

**Issue by pyjauser** [noirello/pyorc#24]
Hi, 

I tried few different python related ORC packages and found pyorc is best. I was testing pyorc for both positive scenario and negative scenario listed from apache orc project. For negative scenario [files](https://github.com/apache/orc/tree/master/examples/corrupt) I was expecting appropriate exception message but pyorc causes   segmentation fault, Could you please look into this? 

```
import pyorc

with open("./missing_blob_stream_in_string_dict.orc", "rb") as data:
    reader = pyorc.Reader(data)
    for row in reader:
        print(row)
```
> Process finished with exit code 139 (interrupted by signal 11: SIGSEGV)

These are the below files which causes segmentation fault when read by pyorc. Some details taken from apache ORC project which may help:

[missing_blob_stream_in_string_dict.orc](https://github.com/apache/orc/blob/master/examples/corrupt/missing_blob_stream_in_string_dict.orc) => [ORC-591](https://issues.apache.org/jira/browse/ORC-591) => [[C++] Check missing blob stream for StringDictionaryColumnRe…](https://github.com/apache/orc/commit/f349dc65af43911f8b839d25bacbd39bcc3dddd9)

[missing_length_stream_in_string_dict.orc](https://github.com/apache/orc/blob/master/examples/corrupt/missing_length_stream_in_string_dict.orc)  => [ORC-590](https://issues.apache.org/jira/browse/ORC-590) => [[C++] added check for missing LENGTH stream in StringDiction…
](https://github.com/apache/orc/commit/a4d6e845576960f9f55c0a43082a8a6cdecd6338)

[negative_dict_entry_lengths.orc](https://github.com/apache/orc/blob/master/examples/corrupt/negative_dict_entry_lengths.orc) => [ORC-589](https://issues.apache.org/jira/browse/ORC-589) => [[C++] add checks about negative dictionary entry lengths](https://github.com/apache/orc/commit/2b012e5a7beb1b688839566522191a8489adb944)

[stripe_footer_bad_column_encodings.orc](https://github.com/apache/orc/blob/master/examples/corrupt/stripe_footer_bad_column_encodings.orc) => [ORC-580](https://issues.apache.org/jira/browse/ORC-580) => [[C++] Verify ColumnEncodings in StripeFooter](https://github.com/apache/orc/commit/0ba08a092dd5ee30a16f7d09c032f8cb2ccaf041) ([#463](https://github.com/apache/orc/pull/463))

**Comment by noirello** [comment]
Hi,

It looks like all of these fixes will be in the upcoming 1.7.0 version of the ORC C++ Core.
(Even the ORC-589, where the ticket says it's fixed in 1.6.3, but it's not patched in the latest 1.6.5)

**Comment by pyjauser** [comment]
Thank you for looking into this.  After building pyorc with latest apache's orc-master branch, expected exceptions were raised for mentioned [files](https://github.com/apache/orc/tree/master/examples/corrupt).
